src/unit_tests/shared/test_atomic.c

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2015-2021, Wazuh Inc.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <string.h>
+
+#include "../../headers/atomic.h"
+#include "../wrappers/posix/pthread_wrappers.h"
+
+atomic_int_t test_variable = ATOMIC_INT_INITIALIZER(0);
+
+
+static int setup_variable(void **state) {
+    expect_function_call(__wrap_pthread_mutex_lock);
+    expect_function_call(__wrap_pthread_mutex_unlock);
+    atomic_int_set(&test_variable, 1231);
+    return 0;
+}
+
+static int teardown_variable(void **state) {
+    expect_function_call(__wrap_pthread_mutex_lock);
+    expect_function_call(__wrap_pthread_mutex_unlock);
+    atomic_int_set(&test_variable, 0);
+    return 0;
+}
+
+
+static void test_atomic_int_get(void **state) {
+    expect_function_call(__wrap_pthread_mutex_lock);
+    expect_function_call(__wrap_pthread_mutex_unlock);
+    int ret = atomic_int_get(&test_variable);
+
+    assert_int_equal(ret, 1231);
+    assert_int_equal(test_variable.data, 1231);
```

Comment:
Testing the value saved inside `test_variable` doesn't make much sense IMO, we are testing the `atomic_int_get` interface, we know the value we expect from the created variable and what we expect as a result of calling `atomic_int_get` on it, we don't care what inner changes are made to `test_variable` as long as the result we get is the one we know to be correct.